### PR TITLE
[MO] - [connects2i] -> with pull-secret

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Enhancement / new feature

### Description

This PR adding support to use external registry and pull image by the KafkaConnectS2I image stream.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: development-docs/TESTING.md [hunk 1]
```diff
@@ -228,6 +228,10 @@ To set custom Kafka version in system tests you need to set the environment vari
 
 To set the log level of Strimzi for system tests you need to set the environment variable `STRIMZI_DEFAULT_LOG_LEVEL` with one of the following values: `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE`.
 
+## Using private registries
+
+If you want use private registries, before executing the tests you have to create secret with the name `pull-secret` to be able pull image.
```
Comment (scholzj) [inline development-docs/TESTING.md:233]:
```suggestion
If you want use private registries, before executing the tests you have to create secret with the name `pull-secret` with the container registry credentials to be able pull images.
```

Comment (see-quick) [reply development-docs/TESTING.md]: Changed thanks

**Comment by see-quick** [comment]
Already tested locally with this specific scenario.

File: development-docs/TESTING.md [hunk 2]
```diff
@@ -228,6 +228,10 @@ To set custom Kafka version in system tests you need to set the environment vari
 
 To set the log level of Strimzi for system tests you need to set the environment variable `STRIMZI_DEFAULT_LOG_LEVEL` with one of the following values: `ERROR`, `WARNING`, `INFO`, `DEBUG`, `TRACE`.
 
+## Using private registries
+
+If you want use private registries, before executing the tests you have to create secret with the name `pull-secret` with the container registry credentials to be able pull images.
```
Comment (Frawless) [inline development-docs/TESTING.md:233]: Wouldn't be better to specify pull-secret name by env variable?

File: development-docs/TESTING.md [hunk 3]
```diff
@@ -224,6 +224,11 @@ If you want to use your own images with a different tag or from a different repo
 
 To set custom Kafka version in system tests you need to set the environment variable `ST_KAFKA_VERSION` to one of the values in [kafka-versions](kafka-versions.yaml).
 
+#### Using private registries
+
+If you want use private registries, before executing the tests you have to create secret and then specify name of the created secret in env variable called
+`STRIMZI_IMAGE_PULL_SECRET` with the container registry credentials to be able pull images. Note that secret has to be created in `default` namespace.
```
Comment (scholzj) [inline development-docs/TESTING.md:230]: I wonder if the name is confusing given the existence of `STRIMZI_IMAGE_PULL_SECRETS` in Cluster Operator.

Comment (see-quick) [reply development-docs/TESTING.md]: I have changed it to -> `SYSTEM_TEST_STRIMZI_IMAGE_PULL_SECRET` as @Frawless suggested.

**Pull request merged**